reconcile/aws_iam_password_reset.py

```diff
@@ -34,40 +36,66 @@ def account_in_roles(roles: Iterable[Mapping[str, Any]], aws_account: str) -> bo
     return False
 
 
+class AwsProfileToReset(BaseModel):
+    account: Mapping[str, Any]
+    user_name: str
+    state_key: str
+
+
 @defer
 def run(dry_run, defer=None):
     accounts = queries.get_aws_accounts(reset_passwords=True)
     settings = queries.get_app_interface_settings()
-    roles = queries.get_roles(aws=True)
+    roles = queries.get_roles(aws=True, saas_files=False)
     state = init_state(integration=QONTRACT_INTEGRATION)
     defer(state.cleanup)
 
+    accounts_to_reset: list[AwsProfileToReset] = []
+
     for a in accounts:
         account_name = a["name"]
         reset_passwords = a.get("resetPasswords")
         if not reset_passwords:
             continue
-        with AWSApi(1, [a], settings=settings) as aws_api:
-            for r in reset_passwords:
-                user_name = r["user"]["org_username"]
-                request_id = r["requestId"]
-                state_key = f"{account_name}/{user_name}/{request_id}"
-                if state.exists(state_key):
-                    continue
-
-                role = get_roles(roles, user_name)
-                if not role:
-                    logging.error(f"Expected a role to be found with name {user_name}")
-                    sys.exit(1)
-
-                if not account_in_roles(role["roles"], account_name):
-                    logging.error(f"User {user_name} is not in account {account_name}")
-                    sys.exit(1)
-
-                logging.info(["reset_password", account_name, user_name])
-                if dry_run:
-                    continue
-
-                aws_api.reset_password(account_name, user_name)
-                aws_api.reset_mfa(account_name, user_name)
-                state.add(state_key)
+
+        for r in reset_passwords:
+            user_name = r["user"]["org_username"]
+            request_id = r["requestId"]
+            state_key = f"{account_name}/{user_name}/{request_id}"
+            if state.exists(state_key):
+                continue
+
+            role = get_roles(roles, user_name)
+            if not role:
+                logging.error(f"Expected a role to be found with name {user_name}")
+                sys.exit(1)
+
+            if not account_in_roles(role["roles"], account_name):
+                logging.error(f"User {user_name} is not in account {account_name}")
+                sys.exit(1)
+
+            accounts_to_reset.append(
+                AwsProfileToReset(
+                    **{
+                        "account": a,
+                        "user_name": user_name,
+                        "state_key": state_key,
+                    }
+                )
+            )
+
+    for a in accounts_to_reset:
+        with AWSApi(1, [a.account], settings=settings) as aws_api:
```

Comment:
wouldn't this establish a new AWS client for each user password to reset, even if all are in the same AWS account?

Reply:
Do you reset accounts for others? :smiling_imp:

Reply:
thanks!